src/cosalib/builds.py

```diff
@@ -62,12 +62,23 @@ def _path(self, path):
     def has(self, build_id):
         return any([b['id'] == build_id for b in self._data['builds']])
 
-    def is_empty(self):
-        return len(self._data['builds']) == 0
+    def is_empty(self, basearch=None):
+        if not basearch:
+            return len(self._data['builds']) == 0
+        # If the user specified an arch then let's return False
+        # only if there exists a build for that architecture
+        for build in self._data['builds']:
+            if basearch in self.get_build_arches(build['id']):
+                return False
+        return True
 
-    def get_latest(self):
-        # just let throw if there are none
-        return self._data['builds'][0]['id']
+    def get_latest(self, basearch=None):
+        if not basearch:
+            # just assume caller wants build dir for current arch
+            basearch = get_basearch()
+        for build in self._data['builds']:
+            if basearch in self.get_build_arches(build['id']):
+                return build['id']
```

Comment:
Hmm, this too I don't think is intuitive behaviour. I think we should always return the latest build regardless of whether it has that basearch or not, and have the logic to check if the build exists for the target arch in the latest build live in `buildprep`.

Reply:
updated to make it a separate function if that sufficiently addresses the concern here.